src/Tools/dotnet-monitor/ConfigurationTokenParser.cs

```diff
@@ -119,11 +119,12 @@ public bool TryCloneSettings(object? originalSettings, ref object? settings)
             return true;
         }
 
-        public static IEnumerable<PropertyInfo> GetPropertiesFromSettings(object? settings, Predicate<PropertyInfo>? predicate = null) =>
-            settings?.GetType()
-                .GetProperties(BindingFlags.Public | BindingFlags.Instance)
-                .Where(p => p.PropertyType == typeof(string) && (predicate?.Invoke(p) ?? true)) ??
-            Enumerable.Empty<PropertyInfo>();
+        public static IEnumerable<PropertyInfo> GetPropertiesFromSettings(Type type, Predicate<PropertyInfo>? predicate = null)
+        {
+            return type.GetProperties(BindingFlags.Public | BindingFlags.Instance)
```

Comment:
Can you clarify why this works correctly for Type.GetProperties but not object?.GetType().GetProperties?